compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1421,7 +1423,12 @@ private STNode parseTypeDescriptor(SyntaxKind tokenKind) {
                 return parseNilTypeDescriptor();
             default:
                 STToken token = peek();
-                Solution solution = recover(token, ParserRuleContext.TYPE_DESCRIPTOR);
+                Solution solution;
+                if (token.kind == SyntaxKind.EOF_TOKEN) {
+                    solution = recover(token, ParserRuleContext.SIMPLE_TYPE_DESCRIPTOR);
+                } else {
+                    solution = recover(token, ParserRuleContext.TYPE_DESCRIPTOR);
+                }
```

Comment:
ParserErrorHandle should handle this IMO

Reply:
Without above modification `test("3 is", "is-expr/is_expr_assert_04.json");` fails.  In error recovery for EOF, we directly insert current ctx. For `TYPE_DESCRIPTOR` ctx, getExpectedTokenKind(currentCtx);` returns a `NONE` syntax kind. 
So I ended up getting a `java.lang.StackOverflowError` as `return parseTypeDescriptor(solution.tokenKind);` line calls repeatedly. `test("3 is ;", "is-expr/is_expr_assert_04.json");` Passes as there's a semicolon next to 'is' keyword

Reply:
I see..
What if we update the `getExpectedTokenKind()` method to return a `SIMPLE_TYPE`, for `TYPE_DESCRIPTOR` context?

Reply:
yeah. it works. let's do that way.